packages/engine-server/src/enginev2.ts

```diff
@@ -589,13 +571,86 @@ export class DendronEngineV2 implements DEngine {
     }
     // TODO: Add another check to see if backlinks have changed
 
-    return (
-      cachedPreview.updated >=
-      NoteUtils.getLatestUpdateTimeOfPreviewNoteTree({
-        rootNote: note,
-        notes: this.notes,
+    const visitedIds = new Set<string>();
+    return this._isCachedPreviewUpToDate({
```

Comment:
Are the changes to this class bug fixes for regressions from refactoring?

Reply:
Not a bug fix; I just wanted to keep it consistent with v3 (even though we're gonna remove this shortly) and it didn't take a long time to change